benchmark/BenchmarkMain.cpp

```diff
@@ -218,4 +206,99 @@ int main() {
+    records.getGroups();
+  const std::vector<BenchmarkRecords::RecordTable>& recordTables =
+    records.getTables();
+
+  // Creating the json object. We actually don't want BenchmarkRecords to
+  // be serilized, because that is the management class for measured
+  // benchmarks. We just want the measured benchmarks.
+  return nlohmann::json{{"singleMeasurements", singleMeasurements},
+    {"recordGroups", recordGroups}, {"recordTables", recordTables}};
+}
+
+void writeJsonToFile(nlohmann::json j, std::string fileName) {
+  // Using this constructor, the file is already opened.
+  std::ofstream myFile(fileName, std::ios::out);
+
+  // Error message, if the file couldn't be opened.
+  if (!myFile.is_open()){
+    std::cerr << "Unable to open file '" << fileName << "'.";
+  }
+
+  // nlohmann already has support for things like this.
+  myFile << j;
+
+  myFile.close();
+}
+
+}
+
+/*
+ * @brief Goes through all types of registered benchmarks, measures their time
+ * and prints their measured time in a fitting format.
+ */
+int main(int argc, char** argv) {
+  // Prints how to use the file correctly and exits.
+  auto printUsageAndExit = [&argv](){
+    std::cerr << "Usage: ./" << argv[0] << " [options]\n"
+      << " --help, -h: Print this help message.\n"
+      << " --print, -p: Roughly prints all benchmarks.\n"
+      << " --json, -j <file>: Writes the benchmarks as json to a file.\n";
+    exit(1);
+  };
+
+  // Calling without using ANY arguments makes no sense.
+  if (argc == 1) {printUsageAndExit();}
+
+  // The filename, should the json option be choosen.
+  std::string jsonFileName = "";
+
+  // Declaring the supported options.
+  boost::program_options::options_description options("options");
+  options.add_options()
+      ("help,h", "Print the help message.")
+      ("print,p", "Roughly prints all benchmarks.")
+      ("json,j", boost::program_options::value<std::string>(&jsonFileName),
+       "Writes the benchmarks as json to a file.")
+  ;
+
+  // Parsing the given arguments.
+  boost::program_options::variables_map vm;
+  boost::program_options::store(
+      boost::program_options::parse_command_line(argc, argv, options), vm);
+  boost::program_options::notify(vm);
+
+  // Did they want the help option?
+  if (vm.count("help")) {printUsageAndExit();}
+
+  // We got at least one argument at this point and all options need all the
+  // benchmarks measured, so time to do that.
+
+  // Measuering the time for all registered benchmarks.
+  // For measuring and saving the times.
+  BenchmarkRecords records;
+ 
+  // Goes through all registered benchmarks and measures them.
+  for (const auto& benchmarkFunction: BenchmarkRegister::getRegisteredBenchmarks()) {
+    benchmarkFunction(&records);
+  }
+
+  // Actually processing the arguments.
+  if (vm.count("print")) {printBenchmarkRecords(records);};
+  
+  if (vm.count("json")) {
+    writeJsonToFile(benchmarksToJson(records), jsonFileName);
+  };
```

Comment:
I don't know how you feel, but for me as a modern human this is much more readable than `getop("h:x:fst:")`
which then segfaults because everybody knows that a `:` is missing after `s`:)

Reply:
I 100% agree.